testing/src/main/java/io/grpc/testing/FakeNameResolverProvider.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.testing;
+
+import com.google.common.collect.ImmutableList;
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.ExperimentalApi;
+import io.grpc.NameResolver;
+import io.grpc.NameResolverProvider;
+import io.grpc.NameResolverRegistry;
+import io.grpc.Status;
+import java.net.SocketAddress;
+import java.net.URI;
+
+/** A name resolver to always resolve the given URI into the given address. */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/8636")
+public final class FakeNameResolverProvider extends NameResolverProvider {
+
+  /**
+   * Register a new resolver.
+   *
+   * @param targetUri The URI to resolve when requested.
+   * @param address The address to return for the target URI.
+   */
+  public static final void register(String targetUri, SocketAddress address) {
```

Comment:
I think we should delete this method and instead expose the constructor. This method is a bad pattern. After a test using this is over, it should really deregister the provider. Conceptually (but maybe using `@After`):

```
provider = new FakeNameResolverProvider(URI.create(targetUri), address);
NameResolverRegistry.getDefaultRegistry().register(provider);
try {
  ...
} finally {
  NameResolverRegistry.getDefaultRegistry().deregister(provider);
}
```

I don't think this class needs to make that try-finally _easier_, but it shouldn't encourage being used in a register-only fashion.

Reply:
Good point, fixed.